Mm 51550 - New Channel Modal is now Work Templates Modal

neallred opened the pull request:
#### Summary
Move new channel modal into work templates modal and replace everywhere that uses new channel modal with work templates modal.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-51550

#### Screenshots
![removed-work-template-item](https://user-images.githubusercontent.com/13738432/234898344-7d816c9b-c070-48d3-9ea1-6fdaa8802941.png)

![combined-modal-with-tip](https://user-images.githubusercontent.com/13738432/234898408-013f7418-682b-43b2-b9c9-c288f514c01d.png)

![combined-modal-without-tip](https://user-images.githubusercontent.com/13738432/234898475-826cdcd7-7f69-4612-9ab1-e8cdb2a5cb63.png)

![combined-modal-without-tip-work-templates-mode](https://user-images.githubusercontent.com/13738432/234898532-99dcf446-75f2-4faa-a5e6-91759b939d68.png)

![combined-modal-after-first-open](https://user-images.githubusercontent.com/13738432/234898600-0ae51077-2645-457e-8cb8-b63b7c7a1d2b.png)

#### Release Note
```release-note
* Work Templates Modal replaces New Channel Modal
```

annelieseklein commented:
Looking good!

1. there should be a 16% Center channel text line under the tabs 
as seen in design:
<img width="222" alt="Screen Shot 2023-04-27 at 1 01 22 PM" src="https://user-images.githubusercontent.com/29821919/234978185-760654ab-6a62-46eb-a896-1f6bcfaf6fad.png">

2. there should be a background here of 4% Center channel text
as seen in design:
<img width="272" alt="Screen Shot 2023-04-27 at 1 03 08 PM" src="https://user-images.githubusercontent.com/29821919/234978332-aef55dab-aa6c-4477-ac33-05b35ae43bc2.png">

3. the "Try a template" cta should be open sans 12, semibold without a background
as seen in design:
<img width="108" alt="Screen Shot 2023-04-27 at 1 04 21 PM" src="https://user-images.githubusercontent.com/29821919/234978664-21b2fce0-9419-4101-9a88-890fa8202476.png">

4. this text should be 100% center channel text
as seen in design:
<img width="177" alt="Screen Shot 2023-04-27 at 1 05 38 PM" src="https://user-images.githubusercontent.com/29821919/234978799-14825569-0bd9-412e-b546-7c5a5a938d54.png">

5. the unselected tab should be 64% center channel text, semibold
as seen in design:
<img width="137" alt="Screen Shot 2023-04-27 at 1 06 32 PM" src="https://user-images.githubusercontent.com/29821919/234978978-f9f3323e-f99f-4d3f-b437-128c54d1d836.png">

6. "Create a board for this channel" text should be 100% center channel text
7. The height of the modal keeps changing, but shouldn't change from one tab to the next, or from one template category to the next. the height should be consistent for 610px to reduce jumping around. 
8. the "New" tab text should be 32 px from the left hand side, just as the title of the modal is 32 px from the left hand side. those two pieces of text should left align. 
9. the X icon on the modal should be 56% center channel text
10. unless this got changed somewhere, the title of the modal shouldn't change depending on what tab you are on - it should remain "Create a new channel"
11. on the templates tab, the header of the template categories should read "CATEGORIES", not "TEMPLATE"
12. when you land on the modal, the channel input field is set to active, which i think is right, except for when we first show the tour tip. i think the input field should be set to default when the tour tip is showing for the first time.

esethna commented:
@neallred nice work! A minor thing I noticed:

The "try a template" button doesn't work on the first try if there is no channel name entered since it shows the error message on the first click: https://gyazo.com/b27f191dd082e3afdb95b23912763375

The "try a template" button and the "Templates" tab should not initiate the empty channel name error message when clicked

neallred commented:
@annelieseklein addressed all points. For your point that "the X icon on the modal should be 56% center channel text", can you confirm if you are still seeing issues with it? It seems set to that for me locally and I haven't made any changes since you first mentioned it. I'm wondering if maybe you were measuring exact colors and doing that forced it to hover state, which also has a darker color?

neallred commented:
@esethna thanks! I've addressed the issue with try template link not firing on first click.

esethna commented:
Thanks @neallred. It seems the error state is appearing very briefly as the template tab is opened. Both when clicking the tab or the try template link

neallred commented:
I'll take a look!

mattermost-build commented:
E2E tests not automatically triggered, because PR has no approval yet. Please ask a developer to review and then try again to attach the QA label.

neallred commented:
Note for dev review: will add unit tests when I pick this up next, but otherwise this is feature complete and seems to implement what is asked from UX and product.

neallred commented:
@esethna flash of error state should be fixed once mm-cloud-bot posts that the spinwick has updated with the latest commit.

neallred commented on webapp/channels/src/components/modal_tabs/index.tsx:
```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
```
Was able to add navigation and selection through `enter`.

Tried adding for space bar too, but it space bar events weren't even making it to these tabs, even if I used capturing phase. I'm guessing it's because its in the modal. Not planning on adding space bar since I think it would require a lot more investigation of the modal and how/if we can circumvent whatever it is doing.

annelieseklein commented:
Looking good! Just one thing which is the "try a template" blue cta when you first land on the modal still does not look semibold. 

Also i think this is probably a separate issue unrelated to the pr, but the checkbox filled looks wrong compared to what we have in design: https://www.figma.com/file/3WslJ69BxdK1ZtecLzSd5q/Components---Checkbox?node-id=9%3A995&t=UNnq4tGgrGfEl3K5-1

neallred commented:
@annelieseklein good finds!

The link seems to match the size and boldness according to the values in the figma spec. I do agree it seems a bit small. While it doesn't seem super bold, it *does* seem super narrow if I take the bold off. Here's its current properties, probably the easiest thing if we need it to pop more is to increase its font size to 13px. What do you think?

![link-styles](https://user-images.githubusercontent.com/13738432/236555129-f1ee4d73-4052-420a-8d5d-8f2133ae42a8.png)

> the rows of the templates should have a space of 16px in between each other

Fixed. You'll be able to see it next time the bot posts `Mattermost test server updated with ...`

> Also i think this is probably a separate issue unrelated to the pr, but the checkbox filled looks wrong

Yep! Separate from the PR. I'd suggest fixing in a separate PR to not make this one bigger.

annelieseklein commented:
> @annelieseklein good finds!
> 
> The link seems to match the size and boldness according to the values in the figma spec. I do agree it seems a bit small. While it doesn't seem super bold, it _does_ seem super narrow if I take the bold off. Here's its current properties, probably the easiest thing if we need it to pop more is to increase its font size to 13px. What do you think?


ok all good - let's just leave it as is! :)

neallred commented on webapp/channels/src/components/work_templates/components/channel_only_footer.tsx:
```diff
@@ -0,0 +1,151 @@
+    cancel?: boolean;
+    confirm?: boolean;
+}
+const primaryButton = `
```
Added a scss file and used the mixins that were available. It still copies some styles from the generic modal for the footer buttons, but I don't want to spend time extracting that out to something more reusable since that modal is being deprecated in favor of the MUI-based modal. [Briefly tried](https://github.com/mattermost/mattermost-server/pull/23189/commits/d67bf7f6689c00105278db9d3a464489c9b78998) using the up and coming MUI-based buttons which would have been nice, but they are [not quite ready](https://github.com/mattermost/mattermost-server/pull/23308) yet.

JulienTant reviewed (approved):
Awesome @neallred! Love to see the new tests!
